### PR TITLE
 Feature/star rating implementation

### PR DESCRIPTION
**Pull request by Dmytro1991ua**
Star Rating implementation. Ability to store user's rate of a certain movie and display in on a screen accordingly and add to Favorites if a rate is 5, created methods to store and retrieve movie rating date from  LocalStorage

### CODE REVIEW

File: src/components/FavoritesBtn.tsx [hunk 1]
```diff
@@ -6,30 +6,40 @@ import '../styles/FavoritesBtn.scss';
 
 type FavoritesBtnProps = {
   movieId: number;
+  isMovieIdInFavorites?: boolean;
 };
 
-const FavoritesBtn: React.FC<FavoritesBtnProps> = ({ movieId }) => {
+const FavoritesBtn: React.FC<FavoritesBtnProps> = ({
+  movieId,
+  isMovieIdInFavorites
```
Comment (oleksandrstarov) [inline src/components/FavoritesBtn.tsx:14]: isnt this same as line 16? what the difference?

File: src/components/FavoritesBtn.tsx [hunk 2]
```diff
@@ -6,30 +6,40 @@ import '../styles/FavoritesBtn.scss';
...
+          src={isMovieIdInFavorites ? heartUnfilledIcon : heartFilledIcon}
+          alt="favorite icon"
+        />
+      ) : (
+        <img
+          className="favorite-icon"
+          src={isFavorite ? heartUnfilledIcon : heartFilledIcon}
+          alt="favorite icon"
+        />
+      )}
```
Comment (oleksandrstarov) [inline src/components/FavoritesBtn.tsx:40]: cant get this part

File: src/service/rating.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+const ratingDataInLocalStorage: { storageKey: string; rating: Array<{}> } = {
+  storageKey: 'movie-rating-data',
+  rating: []
+};
+
+const ratingData = {
+  getMovieRatingFromStorage() {
+    const movieRatingStorage: any = localStorage.getItem(
```
Comment (oleksandrstarov) [inline src/service/rating.tsx:8]: Maybe it is string?

File: src/components/FavoritesBtn.tsx (hunk 1 above)
Comment (Dmytro1991ua) [reply src/components/FavoritesBtn.tsx:14]: After your comment, I realized I gave a misleading name to a state in the first place.  isMovieIdInFavorites here is came from Context and in charge of changing a favorite icon state when a user clicks a star rating icon (when clicking a 5)

File: src/components/FavoritesBtn.tsx (hunk 2 above)
Comment (Dmytro1991ua) [reply src/components/FavoritesBtn.tsx:40]: Here, I was trying to figure out the way how to handle changing a favorite icon state when a user click a 5-star icon (maximum rate). I used a different approach (hopefully now it's correct :) )

File: src/service/rating.tsx (hunk 3 above)
Comment (Dmytro1991ua) [reply src/service/rating.tsx:8]: Thanks for a hint. I have changed a type to string, then a got a message "type of null" for movieRatingStorage is not acceptable to a 'type string', after that a changed a type to a string | null and now it seems to work

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]
Please roll back MovieBanner test deletion

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**